packages/types/src/index.ts

```diff
@@ -87,63 +81,81 @@ export interface MrkdwnElement {
 
 export interface MrkdwnOption {
   text: MrkdwnElement;
-  value?: string;
+  value: string;
   url?: string;
   description?: PlainTextElement;
 }
 
 export interface PlainTextOption {
   text: PlainTextElement;
-  value?: string;
+  value: string;
   url?: string;
   description?: PlainTextElement;
 }
 
-export type Option = MrkdwnOption | PlainTextOption;
-
 export interface Confirm {
-  title?: PlainTextElement;
+  title: PlainTextElement;
   text: PlainTextElement | MrkdwnElement;
-  confirm?: PlainTextElement;
-  deny?: PlainTextElement;
-  style?: 'primary' | 'danger';
+  confirm: PlainTextElement;
+  deny: PlainTextElement;
+  style?: 'primary' | 'danger'; // TODO: factor out into own type
+}
+
+export type Option = MrkdwnOption | PlainTextOption;
+
+export interface DispatchActionConfig {
+  trigger_actions_on?: ('on_enter_pressed' | 'on_character_entered')[];
 }
 
 /*
- * Action Types
+ * Block Elements: https://api.slack.com/reference/block-kit/block-elements
+ * Also known as interactive elements
+ * Exported below as Action
  */
 
+export type KnownAction =
+  Select | MultiSelect | Button | Overflow | Datepicker | Timepicker | RadioButtons | Checkboxes | PlainTextInput;
+
+export interface Action {
+  type: string;
+  /**
+   * @description: A string acting as a unique identifier for a block.
+   */
+  block_id?: string; // TODO: in event payloads, this property will always be present. When defining a block, however,
+  // it is optional. If not defined when the block is created, Slack will auto-assign a block_id for you - thus why it
+  // is always present in payloads. So: how can we capture that? Differentiate between the event payload vs. the block
+  // element?
+}
+
 // Selects and Multiselects are available in different surface areas so I've separated them here
 export type Select = UsersSelect | StaticSelect | ConversationsSelect | ChannelsSelect | ExternalSelect;
 
 export type MultiSelect =
   MultiUsersSelect | MultiStaticSelect | MultiConversationsSelect | MultiChannelsSelect | MultiExternalSelect;
 
-export interface Action {
-  type: string;
-  action_id?: string;
-}
-
 export interface UsersSelect extends Action {
   type: 'users_select';
+  action_id: string;
   initial_user?: string;
-  placeholder?: PlainTextElement;
+  placeholder: PlainTextElement;
```

Comment:
I am sure that this is optional, so this change should be reverted. [The document](https://api.slack.com/reference/block-kit/block-elements#users_select) is wrong! See [a Block Kit Builder example](https://app.slack.com/block-kit-builder/#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Test%20block%20with%20users%20select%22%7D,%22accessory%22:%7B%22type%22:%22users_select%22,%22action_id%22:%22users_select-action%22%7D%7D%5D%7D)

Reply:
You are correct! I was very surprised but confirmed that for all these [Block Elements](https://api.slack.com/reference/block-kit/block-elements), both placeholder and action_id are optional. I am working on a PR to update the docs. Thanks for pointing this out, I have updated these to set placeholder and action_id to optional.